.github/workflows/ci.yml

```diff
@@ -8,18 +8,43 @@ on:
   push:
     branches:
       - main
-      - gnome-43
+      - 'backports/gnome-*'
   pull_request:
     branches:
       - main
-      - gnome-43
+      - 'backports/gnome-*'
   workflow_dispatch:
 
 jobs:
+  select:
+    runs-on: ubuntu-latest
+    outputs:
+      image_id: ${{ steps.select-image.outputs.image_id }}
+    steps:
+      - id: step1
+        if: ${{ ! startsWith(github.event_name, 'pull_request') }}
+        run: echo "target=${{ github.ref_name }}" >> $GITHUB_OUTPUT
+      - id: step2
+        if: ${{ startsWith(github.event_name, 'pull_request') }}
+        run: echo "target=${{ github.base_ref }}" >> $GITHUB_OUTPUT
+      - id: select-image
+        env:
+          TARGET_BRANCH: ${{ steps.step1.outputs.target || steps.step2.outputs.target }}
+        run: |
+          if [[ \
+            ${{ startsWith(env.TARGET_BRANCH, 'backports/') && 1 || 0 }} == 1 \
+          ]]; then
+            IMAGE_ID=$(echo "$TARGET_BRANCH" | sed -e 's,^backports/,,')
+            echo "image_id=$IMAGE_ID" >> $GITHUB_OUTPUT
```

Comment:
(Yes, I arguably made things unnecessarily difficult for myself by naming the branches something _different_ from the container-image tag. But, oh well.)